### DISCUSSION TITLE
Permission errors with external FAT32 filesystem

### DISCUSSION THREAD

**Discussion by jesberpen** [felddy/foundryvtt-docker#233] [Q&A] [answered]
Hi felddy,
I've set my Foundry on a pi using this and has been working great, really awesome work.
I was trying to change my Data folder form the SD on the Pi to a hard drive connected via USB. The drive is recognised and I can create files, edit, etc. no problems there.
In the docker-compose.yml file I change the source for the volume to the root folder in the external drive. But when I do docker-compose up to launch it I get some errors and it fails to launch correctly. In the log files on Portainer I can see some lines with access denied, I am quite new to docker and using the Rapsberry pi, do you know if I need to grant some extra permissions for the container to have access to the drive? I don't think so as I can see the downloaded zip file for foundry that is created in the cache folder. 
I have attached part of the log, I couldn't find a way with portainer to get a log from the start, I think it has some lines from a couple of trials, sorry for that, but hopefully it can help identify the issue.

[_foundry_foundry_1_logs (1).txt](https://github.com/felddy/foundryvtt-docker/files/6982346/_foundry_foundry_1_logs.1.txt)

**Comment by felddy** [comment, marked as answer]
I don't use Portainer, so I'm not sure how much help I can provide.  That said, your problem sounds familiar.  I checked the discussions and didn't see anything related.  

What filesystem are using on the external drive?  I see a lot of MacOS Spotlight cruft in there.
What does your `/etc/fstab` look like?

**Reply by jesberpen** [reply]
Hi, thanks a lot for the quick reply. I've spent the afternoon looking into the issue from my side. The drive was formatted in FAT32, which apparently doesn't accept the chown commands. I made a partition in the drive with ext4 format to be used for this and it worked straight through.
